app/renderer/actions/Session.js

```diff
@@ -214,10 +214,10 @@ export function newSession (caps, attachSessId = null) {
         break;
       case ServerTypes.headspin: {
         const headspinUrl = url.parse(session.server.headspin.webDriverUrl);
-        host = headspinUrl.hostname;
-        port = headspinUrl.port;
-        path = headspinUrl.pathname;
-        https = headspinUrl.protocol === 'https:';
+        host = session.server.headspin.hostname = headspinUrl.hostname;
+        port = session.server.headspin.port = headspinUrl.port;
+        path = session.server.headspin.path = headspinUrl.pathname;
+        https = session.server.headspin.ssl = headspinUrl.protocol === 'https:';
```

Comment:
Do us other cloud providers need to set path too?

Reply:
I'll take a look at other vendors. 👍 
I wondered the path was intentional.

I'll ping proper persons if others also need to consider here.
(And change as another PRs)

Reply:
https://github.com/appium/appium-desktop/compare/master...KazuCocoa:update-session-server-vendors?expand=1
I'll work for it after this PR

Reply:
@KazuCocoa we can let them take care of it themselves. I just wanted to make sure it wasn't breaking anything, which I don't think it will.

Reply:
👍

Yeah, I'm thinking to create the PR and mention them if they would like to follow it